docker/build/cudaq.wheel.Dockerfile

```diff
@@ -27,10 +26,24 @@ ADD "$workspace" "$destination"
 ARG python_version=3.10
 RUN echo "Building wheel for python${python_version}." \
     && cd cuda-quantum && python=python${python_version} \
-    && $python -m pip install cmake auditwheel \
-    && CUDAQ_BUILD_SELFCONTAINED=ON $python -m build --wheel \
+    && $python -m pip install --no-cache-dir \
+        cmake auditwheel \
+        cuquantum-cu11==23.6.0 \
+    && cuquantum_location=`$python -m pip show cuquantum-cu11 | grep -e 'Location: .*$'` \
+    && export CUQUANTUM_INSTALL_PREFIX="${cuquantum_location#Location: }/cuquantum" \
+    && ln -s $CUQUANTUM_INSTALL_PREFIX/lib/libcustatevec.so.1 $CUQUANTUM_INSTALL_PREFIX/lib/libcustatevec.so \
+    && ln -s $CUQUANTUM_INSTALL_PREFIX/lib/libcutensornet.so.2 $CUQUANTUM_INSTALL_PREFIX/lib/libcutensornet.so \
+    &&  SETUPTOOLS_SCM_PRETEND_VERSION=${CUDA_QUANTUM_VERSION:-0.0.0} \
+        CUDAQ_BUILD_SELFCONTAINED=ON \
+        CUDACXX="$CUDA_INSTALL_PREFIX/bin/nvcc" CUDAHOSTCXX=$CXX \
+        $python -m build --wheel \
     && LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$(pwd)/_skbuild/lib" \
-        $python -m auditwheel -v repair dist/cuda_quantum-*linux_*.whl
+        $python -m auditwheel -v repair dist/cuda_quantum-*linux_*.whl \
+            --exclude libcustatevec.so.1 \
+            --exclude libcutensornet.so.2 \
+            --exclude libcublas.so.11 \
+            --exclude libcublasLt.so.11 \
+            --exclude libzstd.so.1
```

Comment:
Q: libcusolver is a dependency of libcutensornet, but I don't see it listed here?

Reply:
Nothing in the build right now uses cutensornet, so the --exclude libcutensornet.so.2 is actually the spare one listed here. I just checked what got included in the wheel and listed what to exclude based on that. Since nothing was using cutensornet, I didn't see any libcusolver dependency either. But good to know - I still need to think about how to best make the tensornet backend available, so that will be relevant once we set that up.